packages/gatsby-plugin-treat/gatsby-node.js

```diff
@@ -0,0 +1,31 @@
+const TreatPlugin = require('treat/webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+const isInDevelopment = (stage) => stage === 'develop' || stage === 'develop-html';
+
+exports.onCreateBabelConfig = ({stage, actions}) => {
+  if (isInDevelopment(stage)) {
+    actions.setBabelPlugin({
+      name: `babel-plugin-treat`,
+    })
+  }
+}
+
+exports.onCreateWebpackConfig = ({stage, actions}) => {
+  if (isInDevelopment(stage)) {
+    actions.setWebpackConfig({
+      plugins: [
+        new TreatPlugin()
+      ]
+   });
+  } else {
+    actions.setWebpackConfig({
+      plugins: [
+        new TreatPlugin({
+          outputLoaders: [MiniCssExtractPlugin.loader]
+        }),
+        new MiniCssExtractPlugin()
+      ]
+   });
+  }
```

Comment:
This is where I am curious if the difference in webpack config could lead to differences between `develop` and `build` styles.  I have had this problem with CSS Modules and think treat could solve that, but I am not sure what the different behavior under the hood is here.

Reply:
What differences do mean specifically? 
Also, I think @jameslnewell has parked this PR so feel free to take it on if you want to.

Reply:
Yep sorry, I've had to put the PR down for now so you're welcome to pick it up!

While writing tests I realised with my current implementationthere was a problem with the generated code when using the `gatsby build` command and I struggled to get `treat` configured correctly with SSR.

I've pushed up my in-progress work if you'd like to continue it!